fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -716,20 +719,20 @@ private static void checkLinkForLdpResourceCreation(final String link) {
      * @return the sha1 checksum value
      * @throws InvalidChecksumException if an unsupported digest is used
      */
-    private static String parseDigestHeader(final String digest) throws InvalidChecksumException {
+    private static Collection<String> parseDigestHeader(final String digest) throws InvalidChecksumException {
         try {
             final Map<String,String> digestPairs = RFC3230_SPLITTER.split(nullToEmpty(digest));
-            final boolean checksumTypeIncludeSHA1 = digestPairs.keySet().stream().anyMatch("sha1"::equalsIgnoreCase);
+            final boolean isSupportedAlgorithm = digestPairs.keySet().stream().anyMatch(
```

Comment:
If I'm reading this right, then any supported checksum in the request header will match.  Further down a few lines, we encounter `.filter(entry -> ContentDigest.DIGEST_ALGORITHM.isSupportedAlgorithm(entry.getKey()))`, so any *non* matching algorithm will be dropped?  

The way I interpret the wording of the ticket is that the presence of *any* non-supported algorithm should fail:
>  any user-provided digest algorithms that are not supported by Fedora should respond with a 400 status or some other indicator to the client that the digest can not be used. 

It's not exactly clear to me where that happens.  Would it be possible to add an IT where the user specifies a SHA-1 and a non-supported digest (like a MD4, or SHA-3, or BADBADNOEXIST-257) and verifies the presence of a 400 error, or other indicator?